crates/bevy_ecs/src/system/query.rs

```diff
@@ -1401,13 +1343,13 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> Query<'w, 's, Q, F> {
     /// ```
     #[inline]
     pub fn contains(&self, entity: Entity) -> bool {
-        // SAFETY: NopFetch does not access any members while &self ensures no one has exclusive access
-        unsafe {
-            self.state
-                .as_nop()
-                .get_unchecked_manual(self.world, entity, self.last_run, self.this_run)
-                .is_ok()
-        }
+        let state = self.state.as_nop();
+
+        let result =
+            // SAFETY: NopFetch does not access any members while &self ensures no one has exclusive access
+            unsafe { state.get_unchecked_manual(self.world, entity, self.last_run, self.this_run) };
```

Comment:
```suggestion
        // SAFETY: NopFetch does not access any members while &self ensures no one has exclusive access
        let result = unsafe { state.get_unchecked_manual(self.world, entity, self.last_run, self.this_run) };
```

It's hard to read when the `SAFETY` comment is in the middle of the statement

Reply:
This isn't valid as-is due to `cargo fmt` rules, but I'll destruct the parameters of this function which should get it short enough to get the desired effect.